source-container-build/app/test_source_build.py

```diff
@@ -0,0 +1,1154 @@
+    origin_path = os.path.join(repos_root, REPO_NAME)
+    os.mkdir(origin_path)
+    cmds = [
+        ["git", "init"],
+        ["git", "config", "user.name", "tester"],
+        ["git", "config", "user.email", "tester@example.com"],
+        ["git", "add", "README.md", "main.py"],
+        ["git", "commit", "-m", "first commit for testing"],
+    ]
+    with open(os.path.join(origin_path, "README.md"), "w") as f:
+        f.write("Testing repo")
+    with open(os.path.join(origin_path, "main.py"), "w") as f:
+        f.write("import this")
+    for cmd in cmds:
+        subprocess.run(cmd, check=True, cwd=origin_path)
+
+    cloned_path = mkdtemp(prefix="local-clone-", dir=repos_root)
+    subprocess.run(["git", "clone", origin_path, cloned_path], check=True)
+
+    return AppSourceDirs(root_dir=repos_root, origin_dir=origin_path, cloned_dir=cloned_path)
+
+
+def create_bsi_cli_parser():
+    """Helper to verify the BuildSourceImage.sh command line arguments
+
+    The idea is, tests do not expect the order of the arguments specified in the implementation.
+    """
+
+    def _clean_drivers(value: str) -> str:
+        cleaned = []
+        for item in value.split(","):
+            if not item:
+                continue
+            if item != "sourcedriver_extra_src_dir" and item != "sourcedriver_rpm_dir":
+                raise argparse.ArgumentTypeError(item)
+            cleaned.append(item)
+        return ",".join(cleaned)
+
+    parser = argparse.ArgumentParser(description="For verifying bsi command line")
+    parser.add_argument("-D", action="store_true", dest="debug_mode")
+    parser.add_argument("-b", metavar="PATH", dest="base_path")
+    parser.add_argument("-o", metavar="PATH", dest="output_path")
+    parser.add_argument("-s", metavar="PATH", dest="srpms_dir")
+    parser.add_argument("-e", metavar="PATH", action="append", dest="extra_src_dirs")
+    parser.add_argument("-d", metavar="DRIVERS", dest="drivers", type=_clean_drivers)
+
+    return parser
+
+
+def create_fake_dep_packages(cachi2_output_dir: str, deps: list[str]) -> None:
+    """Create fake prefetched dependency packages
+
+    :param str cachi2_output_dir: path to cachi2 output directory.
+    :param deps: list of dependency packages. Each of them is in form package_manager/package,
+        e.g. pip/requests-1.0.0.tar.gz
+    :type cachi2_output_dir: list[str]
+    """
+    for package in deps:
+        # pip/, npm/, or gomod/.../.../.../...
+        pkg_mgr_path, filename = os.path.split(package)
+        packages_dir = os.path.join(cachi2_output_dir, "deps", pkg_mgr_path)
+        os.makedirs(packages_dir, exist_ok=True)
+        if filename.endswith(".tar.gz") or filename.endswith(".tgz"):
+            with tarfile.open(os.path.join(packages_dir, filename), "w:gz"):
+                pass
+        elif filename.endswith(".zip"):
+            with zipfile.ZipFile(os.path.join(packages_dir, filename), "w"):
+                pass
+        else:
+            pass
```

Comment:
It looks like this mocking function ignores files that don't end with .tar.gz, .tgz or .zip?